server/tilesource/mapniksource.py

```diff
@@ -409,14 +408,100 @@ def getTile(self, x, y, z, **kwargs):
         pilimg = PIL.Image.frombytes('RGBA', (img.width(), img.height()), img.tostring())
         return self._outputTile(pilimg, TILE_FORMAT_PIL, x, y, z, **kwargs)
 
+    @staticmethod
+    def _proj4Proj(proj):
+        """
+        Return a pyproj.Proj based on either a binary or unicode string.
+
+        :param proj: a binary or unicode projection string.
+        :returns: a proj4 projection object.  None if the specified projection
+            cannot be created.
+        """
+        if isinstance(proj, six.binary_type):
+            proj = proj.decode('utf8')
+        if not isinstance(proj, six.text_type):
+            return
+        if proj.lower().startswith('proj4:'):
+            proj = proj.split(':', 1)[1]
```

Comment:
Why do we have the "proj4:" prefix as an option? I think most geospatial software expects the proj4 string as is.

Reply:
You don't need the `proj4:` prefix if the string starts with `+proj=`.  Proj4 strings don't have to start in any consistent manner (they can start with `+proj=`, `+init=`, `+units=`, etc.).  By adding an extra optional prefix of `proj4:`, I can assume the rest is a proj4 string.

Instead, I could try to generate a `pyproj.Proj` from the string, and, if it throws an exception, assume that it is some other unit that is elsewhere handled.  This means that we would hide pyproj.Proj exceptions, since we wouldn't know if the error occurred because of an invalid proj4 string or a unit that was not otherwise handled.

Reply:
Ok, I think as long as we have the option to directly pass the proj4 string I am happy.